### DISCUSSION TITLE
Run websocket server on same port as other HTTP serving

### DISCUSSION THREAD

**Discussion by khimaros** [dani-garcia/vaultwarden#2917] [Ideas]
This is currently blocked on upstream Rocket feature which has seen some recent progress: https://github.com/SergioBenitez/Rocket/issues/90

**Comment by BlackDex** [comment]
I Think this is going to take a while when reading that ticket.
Luckily we can use reverse proxies to solve this shortcoming for now.

**Comment by BlackDex** [comment]
Going to close this as an issue.
I have added it to the META Features request #246 .

**Comment by fidelix** [comment]
BTW, this is now possible and is available upstream.

**Reply by BlackDex** [reply]
See #3404

**Reply by binaryben** [reply]
Neat, came to the say the same and it's already done! Would anyone be kind enough please to let me know what changes need to be made when using Caddy as a reverse proxy? Second question - the current config seems to still work, so is there going to be both options for getting websockets to work?

**Reply by BlackDex** [reply]
@binaryben basically remove the `reverse_proxy /notifications/hub <SERVER>:3012` line should do it.
You can also set `WEBSOCKET_ENABLED` to `false`, which prevents a separate thread from being started.

**Reply by binaryben** [reply]
Thanks @BlackDex. Is that in the current vaultwarden/server:latest docker image? I'll try it and report back once my Postgres service is back online if it is 😬

**Reply by BlackDex** [reply]
You need to use `:testing` for this to work.
